osf/models/node.py

```diff
@@ -391,6 +395,24 @@ def parent_node(self):
                 return parent
         return None
 
+    @property
+    def storage_limit_status(self):
+        """ This should indicate if a node is at or over a certain storage threshold indicating a status. If nodes have
+        a custom limit this should indicate that."""
+
+        limits = dict([(item.name, item.value) for item in settings.StorageLimits])
+        if self.custom_storage_usage_limit_public:
+            limits['OVER_PUBLIC'] = self.custom_storage_usage_limit_public
+            warning_limit = self.custom_storage_usage_limit_public * settings.STORAGE_WARNING_THRESHOLD
+            limits['APPROACHING_PUBLIC'] = warning_limit
+        if self.custom_storage_usage_limit_private:
+            limits['OVER_PRIVATE'] = self.custom_storage_usage_limit_private
+            warning_limit = self.custom_storage_usage_limit_private * settings.STORAGE_WARNING_THRESHOLD
+            limits['APPROACHING_PRIVATE'] = warning_limit
+
+        limits = enum.IntEnum('StorageLimitsWithCustomValues', [(key, value) for key, value in limits.items()])
+        return max(limit for limit in limits if limit.value <= self.storage_usage)
```

Comment:
As suggested below, we shouldn't be encoding the actual limit values in the enum. If we go with a more abstract approach, we can hide this logic away as a class function on the Enum:
```
@classmethod
fromNodeUsage(cls, usage_bytes, private_limit=None, public_limit=None):
```

Where None implies a NULL database entry and, thus, the default values.